Automattic/vip-cli
# Reconnect to job if disconnected

## Description

This fix the problem with reconnections

There are still some bugs:
- When disconnected, an error is written to the console, do we want this?
- Another bug is in the test steps bellow

## Steps to Test

Outline the steps to test and verify the PR here.

Example:

1. Check out PR
2. Make sure all APIs are up with all necessary things to make WP-CLI work in localhost
3. Run `npm link`
4. Run `API_HOST=http://localhost:4000 vip wp shell`
5. Interact with the shell
6. Cause the public API to reload by changing something on the code (can be a console log or something)
7. You'll see an error `Error: Connection aborted` alongside a `lost connection` <- this is added just for dev purpose and should be removed once we are ready
8. Once the API is up again, you'll see a `reconnected` in the console, this also is added just for dev purpose and will be removed afterwards
9. Continue your interaction with the shell
10. You'll receive an `invalid command, please pass a valid WP CLI command.` error. **This is a bug** we still need to see why it's raising. It looks like it tries to send another request even if we are in shell mode.

Fix #417
